### PR TITLE
Add Purescript language server

### PR DESCRIPTION
**Pull request by AntoineGagne**
The `root_dir` supports both `pulp` and `spago`.

### CODE REVIEW

File: lua/nvim_lsp/purescriptls.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    local install_info = installer.info()
+    if install_info.is_installed then
+      if type(new_config.cmd) == 'table' then
+        new_config.cmd[1] = install_info.binaries[bin_name]
+      else
+        new_config.cmd = {install_info.binaries[bin_name]}
+      end
+    end
+  end;
+  -- on_attach = function(client, bufnr) end;
```
Comment (h-michael) [inline lua/nvim_lsp/purescriptls.lua]:
```suggestion
```

File: lua/nvim_lsp/purescriptls.lua [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  docs = {
```
Comment (h-michael) [inline lua/nvim_lsp/purescriptls.lua]:
```suggestion
  docs = {
  vscode = "nwolverson.ide-purescript";
```

**Comment by AntoineGagne** [comment]
I made the changes according to the comment. However, `luarocks` seems to be having some trouble downloading `luacheck`. I am not sure what is happening (especially since I haven't made any changes to the CI definition).

**Comment by h-michael** [comment]
@AntoineGagne ty :)

**Pull request merged**